SQL/New_patches/2018-09-07_EEG_tables.sql

```diff
@@ -0,0 +1,264 @@
+    FOREIGN KEY (`FileType`)
+    REFERENCES `ImagingFileTypes` (`type`),
+  CONSTRAINT `FK_physiological_modality_PhysiologicalModalityID`
+    FOREIGN KEY (`PhysiologicalModalityID`)
+    REFERENCES `physiological_modality` (`PhysiologicalModalityID`),
+  CONSTRAINT `FK_physiological_output_type_PhysiologicalOutputTypeID`
+    FOREIGN KEY (`PhysiologicalOutputTypeID`)
+    REFERENCES `physiological_output_type` (`PhysiologicalOutputTypeID`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
+
+
+
+-- Create a physiological_parameter_file table that will store all JSON
+-- information that accompanies the BIDS physiological dataset
+CREATE TABLE `physiological_parameter_file` (
+  `PhysiologicalParameterFileID` INT(10) UNSIGNED NOT NULL  AUTO_INCREMENT,
+  `PhysiologicalFileID`          INT(10) UNSIGNED NOT NULL,
+  `ParameterTypeID`              INT(10) UNSIGNED NOT NULL,
+  `InsertTime`                   TIMESTAMP        NOT NULL  DEFAULT CURRENT_TIMESTAMP,
+  `Value`                        VARCHAR(255),
+  PRIMARY KEY (`PhysiologicalParameterFileID`),
+  CONSTRAINT `FK_physiological_file_PhysiologicalFileID`
+    FOREIGN KEY (`PhysiologicalFileID`)
+    REFERENCES `physiological_file` (`PhysiologicalFileID`)
+    ON DELETE CASCADE,
+  CONSTRAINT `FK_parameter_type_ParameterTypeID`
+    FOREIGN KEY (`ParameterTypeID`)
+    REFERENCES `parameter_type` (`ParameterTypeID`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
+
+
+
+-- Create a physiological_status_type table
+CREATE TABLE `physiological_status_type` (
+  `PhysiologicalStatusTypeID` INT(5)      UNSIGNED NOT NULL AUTO_INCREMENT,
+  `ChannelStatus`             VARCHAR(10)          NOT NULL UNIQUE,
+  PRIMARY KEY (`PhysiologicalStatusTypeID`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
+
+
+
+-- Create a physiological_channel_type table
+CREATE TABLE `physiological_channel_type` (
+  `PhysiologicalChannelTypeID` INT(5)       UNSIGNED NOT NULL      AUTO_INCREMENT,
+  `ChannelTypeName`            VARCHAR(255)          NOT NULL      UNIQUE,
+  `ChannelDescription`         VARCHAR(255)          DEFAULT NULL,
+  PRIMARY KEY (`PhysiologicalChannelTypeID`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
+
+
+
+-- Create a physiological_channel table that will store all information from
+CREATE TABLE `physiological_channel` (
+  `PhysiologicalChannelID`     INT(10)    UNSIGNED NOT NULL      AUTO_INCREMENT,
+  `PhysiologicalFileID`        INT(10)    UNSIGNED NOT NULL,
+  `PhysiologicalChannelTypeID` INT(5)     UNSIGNED NOT NULL,
+  `PhysiologicalStatusTypeID`  INT(5)     UNSIGNED DEFAULT NULL,
+  `InsertTime`                 TIMESTAMP           NOT NULL      DEFAULT CURRENT_TIMESTAMP,
+  `Name`                       VARCHAR(50)         NOT NULL,
+  `Description`                VARCHAR(255)        DEFAULT NULL,
+  `SamplingFrequency`          INT(6)              DEFAULT NULL,
+  `LowCutoff`                  DECIMAL(8,3)        DEFAULT NULL,
+  `HighCutoff`                 DECIMAL(8,3)        DEFAULT NULL,
```

Comment:
@PapillonMcGill I have a user that has float values and the string "Inf" for that field in their channels.tsv file. His file has been created directly from a edf converter so we might end up with other users having that field populated to a number or "Inf".
Not sure what would be the best option here. Maybe convert the field to a varchar so we could store the float and "Inf" value?

What are your thoughts regarding this?

Reply:
After investigation on the subject and discussion with @cmadjar, recommendation is to use 99999.999 to indicate "inf" in the value. Code will need to support the bi-directional conversion.

Reply:
This change is supported by the insertion script now. Currently, this value is not displayed in the Front End but we will keep this in mind. Thanks!